noel/setplot.py

```diff
@@ -0,0 +1,270 @@
+        plotfigure.kwargs = {"figsize": region_dict['figsize']}
+        plotaxes = plotfigure.new_plotaxes()
+        plotaxes.title = "Currents"
+        plotaxes.xlimits = region_dict["xlimits"]
+        plotaxes.ylimits = region_dict["ylimits"]
+        plotaxes.afteraxes = surge_afteraxes
+
+        surgeplot.add_speed(plotaxes, bounds=speed_limits)
+        surgeplot.add_land(plotaxes)
+        plotaxes.plotitem_dict['speed'].amr_patchedges_show = [0] * 10
+        plotaxes.plotitem_dict['land'].amr_patchedges_show = [0] * 10
+    #
+    # Friction field
+    #
+    plotfigure = plotdata.new_plotfigure(name='Friction')
+    plotfigure.show = friction_data.variable_friction and True
+
+    plotaxes = plotfigure.new_plotaxes()
+    plotaxes.xlimits = regions['Atlantic']['xlimits']
+    plotaxes.ylimits = regions['Atlantic']['ylimits']
+    # plotaxes.title = "Manning's N Coefficient"
+    plotaxes.afteraxes = friction_after_axes
+    plotaxes.scaled = True
+
+    surgeplot.add_friction(plotaxes, bounds=friction_bounds, shrink=0.9)
+    plotaxes.plotitem_dict['friction'].amr_patchedges_show = [0] * 10
+    plotaxes.plotitem_dict['friction'].colorbar_label = "$n$"
+
+    #
+    #  Hurricane Forcing fields
+    #
+    # Pressure field
+    plotfigure = plotdata.new_plotfigure(name='Pressure')
+    plotfigure.show = surge_data.pressure_forcing and True
+
+    plotaxes = plotfigure.new_plotaxes()
+    plotaxes.xlimits = regions['Atlantic']['xlimits']
+    plotaxes.ylimits = regions['Atlantic']['ylimits']
+    plotaxes.title = "Pressure Field"
+    plotaxes.afteraxes = surge_afteraxes
+    plotaxes.scaled = True
+    surgeplot.add_pressure(plotaxes, bounds=pressure_limits)
+    surgeplot.add_land(plotaxes)
+
+    # Wind field
+    plotfigure = plotdata.new_plotfigure(name='Wind Speed')
+    plotfigure.show = surge_data.wind_forcing and True
+
+    plotaxes = plotfigure.new_plotaxes()
+    plotaxes.xlimits = regions['Atlantic']['xlimits']
+    plotaxes.ylimits = regions['Atlantic']['ylimits']
+    plotaxes.title = "Wind Field"
+    plotaxes.afteraxes = surge_afteraxes
+    plotaxes.scaled = True
+    surgeplot.add_wind(plotaxes, bounds=wind_limits)
+    surgeplot.add_land(plotaxes)
+
+    # ========================================================================
+    #  Figures for gauges
+    # ========================================================================
+
+    plotfigure = plotdata.new_plotfigure(name='Gauge Surfaces', figno=300,
+                                         type='each_gauge')
+    plotfigure.show = True
+    plotfigure.clf_each_gauge = True
+
+    # Set up for axes in this figure:
+    plotaxes = plotfigure.new_plotaxes()
+    plotaxes.xlimits = [-3, 5]
+    # plotaxes.xlabel = "Days from landfall"
+    # plotaxes.ylabel = "Surface (m)"
+    plotaxes.ylimits = [-2, 3]
+    plotaxes.title = 'Surface'
+
+    # comparison water level files
+    gauges = [np.loadtxt('gauge_data_1.txt'), np.loadtxt('gauge_data_2.txt'),
```

Comment:
If these data files do not exist these plots unfortunately just fail.  We should discuss either putting this in a `try-except` block or automatically downloading them or something else.

Reply:
thanks for the comments, ill look into them! ill also fix the computation halted issue

Reply:
maybe i can edit comparison_gauge_data.py and then add that code to setplot so it can automatically download the files